frame/staking/src/lib.rs

```diff
@@ -745,17 +745,46 @@ enum Releases {
 	V4_0_0,
 	V5_0_0, // blockable validators.
 	V6_0_0, // removal of all storage associated with offchain phragmen.
+	V7_0_0, // keep track of number of nominators / validators in map
 }
 
 impl Default for Releases {
 	fn default() -> Self {
-		Releases::V6_0_0
+		Releases::V7_0_0
 	}
 }
 
 pub mod migrations {
 	use super::*;
 
+	pub mod v7 {
+		use super::*;
+
+		pub fn pre_migrate<T: Config>() -> Result<(), &'static str> {
+			assert!(CurrentValidatorsCount::<T>::get().is_zero(), "CurrentValidatorsCount already set.");
+			assert!(CurrentNominatorsCount::<T>::get().is_zero(), "CurrentNominatorsCount already set.");
+			assert!(StorageVersion::<T>::get() == Releases::V6_0_0);
+			Ok(())
+		}
+
+		pub fn migrate<T: Config>() -> Weight {
+			log!(info, "Migrating staking to Releases::V7_0_0");
```

Comment:
Should you not check that the previous version is `Releases::V7_0_0"`? otherwise, we might execute this multiple times accidentally.

Reply:
or if you are planning on adding this to `Hooks`, make sure to check the version there.

Reply:
I wanted to run the migrations on a call with Zeke and needed to add them to `Hooks`, might as well push them here for you.

Reply:
sure, just push your changes